More information about departments

jenssegers opened the pull request:
Related Implementation PR: https://github.com/teamleadercrm/core/pull/6753
Related E2E PR: https://github.com/teamleadercrm/api-tests/pull/410

Integrators might want to get information about the company of the user to set up an initial account on their end, or for invoicing reasons.

jenssegers committed:
Add more information for departments

marijnvandevoorde commented:
@steftrenson, WDYT: should we add currency here as well while we're at it?

Other than that, looking fine!

steftrenson commented:
@mrnnn You could already do that, although I have no idea on what the timeframe is to actually have a department in a different currency

Commit:
Adds currency to the info & list responses

It's there already in the implementation.

marijnvandevoorde reviewed (dismissed):
Let's implement this.

So far we have id, name & currency

thomasnoe07 commented:
@jenssegers Can we move this further or it should stay open?

luisaterrabrito commented:
Why the same information in the list and info?

jenssegers committed:
:twisted_rightwards_arrows: Merge master

jenssegers committed:
Add iban and bic to departments

dcsg commented:
I agree with @luisabritotl, why do we have the same information in `list` and `info`? Also, why the `website`?

paultimoce commented on apiary.apib:
```diff
@@ -291,7 +291,25 @@ Get a list of departments.
+                                + invoicing
+                        + email: `info@piedpiper.eu` (string)
+                + telephones (array[object])
+                    + type: `phone` (enum[string], required)
```
This should be mobile i think since the corresponding database field is gsm

WouterSioen replied:
only mobile and fax for departments for now

jenssegers replied:
In the UI it's called telephone and fax, hence the phone here. Phone also makes more sense on company level (imo).

WouterSioen replied:
😞

jenssegers commented:
I see no problem in having the same information returned in both the list and info endpoints. It's basically the same information we're returning here as for the contact/company endpoints.

dcsg commented:
@jenssegers why iban, website, bic and that type of information is available in a `list` endpoint? It belongs to the **details** of the `resource` hence in the `info` endpoint. Otherwise, why do we have two endpoints? We could just have the `list` one.

dcsg commented on apiary.apib:
```diff
@@ -308,7 +326,25 @@ Get details for a single department.
+                + number: `092980615` (string, required)
+            + website: `https://piedpiper.com` (string)
             + currency: `EUR` (string)
+            + iban: `BE12123412341234` (string)
```
B/Q: why are we adding this to the root level? This for me seems to be missing a "bank information" object and then having these fields inside that object.

dietrichm replied:
Lots of endpoints are adding IBAN and BIC on the top level right now. This might be a BC breaking improvement that we can release in a next API version.

luisaterrabrito commented:
I don't agree, @jenssegers. I understand more information might be added in the future to the info endpoint, but for now, I find it confusing, I wouldn't know which one to use. Feels like there is no need to have both.

jenssegers commented:
Ok let's reduce the information on the list endpoint then. Easier to add it later than to remove it afterwards. Any other suggestions for fields to drop other than the iban/bic?

luisaterrabrito commented:
I don't know where it is/will be used, but making a comparison with the invoices list that is used in the redesign to show all the invoices, we only need some of the basics, I would remove the addresses, emails, telephones, website or just leave the default ones and show the others in the info. The emails have one that is primary, I guess we could return that one and if the department has more of them show it in the info

luisaterrabrito commented:
Please leave the currency, I think @lowiebenoot might be using it

lowiebenoot commented:
I'm indeed using the currencies on `departments.list`.

WouterSioen commented:
Who's following up to make the specs correct? @jenssegers ?

WouterSioen committed:
Get rid of too much data in departments.list endpoint

We have .info endpoints for that.

WouterSioen committed:
Build main apiary file

WouterSioen reviewed (dismissed):
I cleaned up the .list endpoint and got rid of a lot of fields.

Is this good to go now?

dcsg commented on apiary.apib:
```diff
@@ -291,6 +291,8 @@ Get a list of departments.
                 + id: `67c576e7-7e6f-465d-b6ab-a864f6e5e95b` (string)
                 + name: `Human Resources` (string)
+                + vat_number: `BE0899623035` (string)
+                + website: `https://piedpiper.com` (string)
```
Q: why is this necessary in the `list`? My concern is that this field is not required and I would imagine that most of the times it will be `null`?

WouterSioen replied:
I can imagine that you'd like to show that in an overview of departments?

WouterSioen replied:
Okay, you're right, website is filled in for only 15% of departments and is not really prominent in the UI, so it'll be `null` a lot of the times.

Not sure if this is an issue though? I can still imagine this getting a more prominent place later?

WouterSioen replied:
We could still add the fields at that moment though.

dcsg replied:
The thing is if it's only 15% than the UI will have an empty column 85% of the times will be awkward for the user IMO other than that having the field would be ok ;)

WouterSioen committed:
Also get rid of the website field in the list endpoint

It's not really prominent in the application right now resulting on it
only being filled in for around 15% of departments.

We can always add the field later if it would get a more important place
in our application(s).

WouterSioen commented:
Let's consider this proposal final so we can start on the implementation.

luisaterrabrito commented:
Sorry guys, I thought I was on my PR
